software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino

```diff
@@ -0,0 +1,230 @@
+// HeartRateDetection_V2.0 - BioAmp EXG Pill
+// https://github.com/upsidedownlabs/BioAmp-EXG-Pill
+// https://github.com/upsidedownlabs/Heart-BioAmp-Arduino-Firmware
+
+// Upside Down Labs invests time and resources providing this open source code,
+// please support Upside Down Labs and open-source hardware by purchasing
+// products from Upside Down Labs!
+
+// Copyright (c) 2021 - 2024 Upside Down Labs - contact@upsidedownlabs.tech
+// Copyright (c) 2021 - 2024 Aryan Prakhar - aryanprakhar1010@gmail.com
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+
+#include <math.h>
+
+
+// Define constants
+#define SAMPLE_RATE 125       // Sampling rate in Hz
+#define BAUD_RATE 115200      // Serial communication baud rate
+#define INPUT_PIN A2          // Analog  pin for ECG signal
+#define OUTPUT_PIN 13         // Digital output pin for LED
+#define DATA_LENGTH 26        // Length of data buffer for peak detection
+#define buffer_size 5         // Size of circular arrays for BPM calculation
+
+// Global variables for BPM calculation
+int data_index = 0;
+uint32_t tdiff = 0;           // Time difference between consecutive peaks
+uint8_t BPM = 0;              // Calculated Beats Per Minute

```

Comment:
Please use a standard way of choosing variable names. I can define this under docs if it's not clear to you.

Reply:
I have used lower case the variable to standardized the variables used in this program